### PR TITLE
add gha test.yml file

### PR DESCRIPTION
**Pull request by Xiangs18**

### CODE REVIEW

File: .github/workflows/test.yml
Comment (MrCreosote) [inline .github/workflows/test.yml:1]: Can you add the other GHA workflows? You should be able to just copy them from collections

Comment (MrCreosote) [inline .github/workflows/test.yml:1]: Is there any reason not to delete travis.yml?

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+            minio: '2019-05-23T00-29-34Z'
+            wired_tiger: 'false'
+          - go: '1.12'
+            mongo: 'mongodb-linux-x86_64-3.6.20'
+            minio: '2019-05-23T00-29-34Z'
+            wired_tiger: 'false'
+          - go: '1.12'
+            mongo: 'mongodb-linux-x86_64-3.6.20'
+            minio: '2019-05-23T00-29-34Z'
+            wired_tiger: 'true'
```
Comment (MrCreosote) [inline .github/workflows/test.yml:35]: When you update with mongo 7 you can probably get rid of the 3.6.20 wired_tiger: false matrix entry

File: .github/workflows/test.yml
Comment (MrCreosote) [inline .github/workflows/test.yml:1]: Looks like Codecov or the coverage calculator changed how they count lines in missed branches

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+          # set HOMEDIR
+          export HOMEDIR=`pwd`
+
+          # move to parent dir to install binaries etc
+          cd ..
+
+          # set up jars
+          git clone https://github.com/kbase/jars
+          export JARSDIR=`pwd`/jars/lib/jars/
```
Comment (MrCreosote) [inline .github/workflows/test.yml:55]: Can you add a TODO to switch to the auth shadow jar?

File: go.sum [hunk 3]
```diff
@@ -2,7 +2,6 @@ github.com/aws/aws-sdk-go v1.19.21 h1:xLaPxl8gy0ZSXbc13jsCKIaHD6NiX+2tAQodPSEL5r
 github.com/aws/aws-sdk-go v1.19.21/go.mod h1:KmX6BPdI08NWTb3/sm4ZGu5ShLoqVDhKgpiN924inxo=
 github.com/coreos/go-semver v0.3.0 h1:wkHLiw0WNATZnSG7epLsujiMCgPAc9xhjJ4tgnAxmfM=
 github.com/coreos/go-semver v0.3.0/go.mod h1:nnelYz7RCh+5ahJtPPxZlU+153eP4D4r3EedlOD2RNk=
-github.com/davecgh/go-spew v1.1.0 h1:ZDRjVQ15GmhC3fiQ8ni8+OwkZQO4DARzQgrnXU1Liz8=
```
Comment (MrCreosote) [inline go.sum:5]: This and the deletion below seem weird. How did this happen?

**Review by MrCreosote — COMMENTED** [review]
Looks pretty good

File: go.sum (hunk 3 above)
Comment (Xiangs18) [reply go.sum:5]: I ran `go mod tidy` and then these 2 lines are removed.

Comment (MrCreosote) [reply go.sum:5]: Strange, but it builds so I guess this is ok

Comment (MrCreosote) [reply go.sum:5]: :+1:

File: .github/workflows/test.yml (hunk 1 above)
Comment (Xiangs18) [reply .github/workflows/test.yml:35]: Will do. Also why we test 2 different v3 here?

File: .github/workflows/test.yml (hunk 2 above)
Comment (Xiangs18) [reply .github/workflows/test.yml]: strange, TODO is not highlighted in yml file.

File: .github/workflows/test.yml
Comment (Xiangs18) [reply .github/workflows/test.yml:1]: deleted

Comment (Xiangs18) [reply .github/workflows/test.yml:1]: It seems `release-main.yml` and `manual-main.yml` are not triggered? Do we still need them?

Comment (Xiangs18) [reply .github/workflows/test.yml:1]: What has changed?

Comment (bio-boris) [reply .github/workflows/test.yml:1]: Release main is triggered when you press the "Release" button and draft a release
Manual is for manual builds
We have some docs at https://github.com/kbase/.github

Comment (Xiangs18) [reply .github/workflows/test.yml:1]: @bio-boris Also, Are codecov/project and codecov/patch checks automatically triggered when I activate codecov? Or we need to initiate or configure somewhere?

Comment (bio-boris) [reply .github/workflows/test.yml:1]: @Xiangs18 It looks like its here https://app.codecov.io/github/kbase/blobstore

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+      - name: Run tests
+        shell: bash
+        run: |
+          export GO111MODULE=on
+          go build app/blobstore.go
+          export TCFG=`pwd`/test.cfg
+          BLOBSTORE_TEST_CFG=$TCFG go test -race -coverprofile=coverage.txt -covermode=atomic ./...
+
+      - name: Upload coverage to Codecov
```
Comment (bio-boris) [inline .github/workflows/test.yml:86]: @Xiangs18 This is where you activated codecov

Comment (Xiangs18) [reply .github/workflows/test.yml:86]: yeah, my question is more like:
`codecov/project` and `codecov/patch` checks just come out of box when I activated codecov?

Comment (bio-boris) [reply .github/workflows/test.yml:86]: Sorry, I don't know

File: .github/workflows/test.yml (hunk 1 above)
Comment (MrCreosote) [reply .github/workflows/test.yml:35]: At the time, v12 was in production (I think? Might be v13, but pretty close anyway) and v20 was the latest 3.6 release

File: .github/workflows/test.yml
Comment (MrCreosote) [reply .github/workflows/test.yml:1]: Check out the codecov report, you'll see what I mean. More lines are counted as missed but the actual code is the same

File: .github/workflows/test.yml (hunk 4 above)
Comment (MrCreosote) [reply .github/workflows/test.yml:86]: They do, once you set the coverage file to codecov it does the rest (at least in the kbase org - there might be other setup needed for other orgs)

File: .github/workflows/test.yml
Comment (MrCreosote) [reply .github/workflows/test.yml:1]: :+1:

File: .github/workflows/test.yml (hunk 2 above)
Comment (MrCreosote) [reply .github/workflows/test.yml]: :+1:

File: .github/workflows/test.yml
Comment (MrCreosote) [reply .github/workflows/test.yml:1]: :+1:

**Review by MrCreosote — APPROVED** [review]
LGTM

File: .github/workflows/test.yml (hunk 1 above)
Comment (Xiangs18) [reply .github/workflows/test.yml:35]: 👍

File: .github/workflows/test.yml (hunk 4 above)
Comment (Xiangs18) [reply .github/workflows/test.yml:86]: 👍

File: .github/workflows/test.yml
Comment (Xiangs18) [reply .github/workflows/test.yml:1]: 👍

**Pull request merged**